internal/agent/device/applications/provider/provider.go

```diff
@@ -58,6 +56,325 @@ type ApplicationSpec struct {
+			imageSpec, err := providerSpec.AsImageApplicationProviderSpec()
+			if err != nil {
+				return nil, fmt.Errorf("getting image provider spec: %w", err)
+			}
+
+			// Add base image with PullIfNotPresent policy
+			policy := v1alpha1.PullIfNotPresent
+			targets = append(targets, dependency.OCIPullTarget{
+				Type:       dependency.OCITypeImage,
+				Reference:  imageSpec.Image,
+				PullPolicy: policy,
+				PullSecret: pullSecret,
+			})
+
+			// Add volume artifacts
+			volTargets, err := extractVolumeTargets(imageSpec.Volumes, pullSecret)
+			if err != nil {
+				return nil, fmt.Errorf("extracting volume targets: %w", err)
+			}
+			targets = append(targets, volTargets...)
+
+		case v1alpha1.InlineApplicationProviderType:
+			inlineSpec, err := providerSpec.AsInlineApplicationProviderSpec()
+			if err != nil {
+				return nil, fmt.Errorf("getting inline provider spec: %w", err)
+			}
+
+			appType := lo.FromPtr(providerSpec.AppType)
+			if appType == "" {
+				return nil, fmt.Errorf("appType is required for inline providers")
+			}
+
+			// Extract images from inline content based on app type
+			switch appType {
+			case v1alpha1.AppTypeCompose:
+				// Inline compose specs are already validated by the API
+				spec, err := client.ParseComposeFromSpec(inlineSpec.Inline)
+				if err != nil {
+					return nil, fmt.Errorf("parsing compose spec: %w", err)
+				}
+				for _, svc := range spec.Services {
+					if svc.Image != "" {
+						targets = append(targets, dependency.OCIPullTarget{
+							Type:       dependency.OCITypeImage,
+							Reference:  svc.Image,
+							PullPolicy: v1alpha1.PullIfNotPresent,
+							PullSecret: pullSecret,
+						})
+					}
+				}
+			case v1alpha1.AppTypeQuadlet:
+				spec, err := client.ParseQuadletReferencesFromSpec(inlineSpec.Inline)
+				if err != nil {
+					return nil, fmt.Errorf("parsing quadlet spec: %w", err)
+				}
+				for _, quad := range spec {
+					targets = append(targets, extractQuadletTargets(quad, pullSecret)...)
+				}
+			default:
+				return nil, fmt.Errorf("%w: %s", errors.ErrUnsupportedAppType, appType)
+			}
+
+			// Add volume artifacts
+			volTargets, err := extractVolumeTargets(inlineSpec.Volumes, pullSecret)
+			if err != nil {
+				return nil, fmt.Errorf("extracting volume targets: %w", err)
+			}
+			targets = append(targets, volTargets...)
+
+		default:
+			return nil, fmt.Errorf("unsupported application provider type: %s", providerType)
+		}
+	}
+
+	embeddedTargets, err := collectEmbeddedOCITargets(ctx, readWriter, pullSecret)
+	if err != nil {
+		return nil, fmt.Errorf("collecting embedded OCI targets: %w", err)
+	}
+	targets = append(targets, embeddedTargets...)
+
+	return targets, nil
+}
+
+// collectEmbeddedOCITargets discovers embedded applications and extracts their OCI targets
+func collectEmbeddedOCITargets(ctx context.Context, readWriter fileio.ReadWriter, pullSecret *client.PullSecret) ([]dependency.OCIPullTarget, error) {
```

Comment:
embedded quadlets are missing. Should be realtively easy to adjust this to accommodate searching in the appropriate directories. Look at parseEmbeddedQuadlets